WordPress/build.gradle

```diff
@@ -238,6 +243,33 @@ android {
     }
 }
 
+/// Dynamically add `buildConfigFields` on a given variant/flavor from prefixed properties
+/// (This is used to e.g. add every property prefixed `wp.` in `gradle.properties` as a BuildConfigField in the `wordpress` flavor)
+///
+/// `prefixes` is an array of prefixes to search for, without their `.` dot (e.g. `["wp", "jp"]`).
+/// Properties found with prefixes first in that array will be overriden by the ones found with prefixes later in the array.
+def addBuildConfigFieldsFromPrefixedProperties(variant, properties, prefixes) {
+    // Build a Map of prefixed properties found. Keys in this Map will be lowercased and `.` replaced by `_`
+    def fields_map = [:]
+    prefixes.each { prefix ->
+        properties.each { property ->
+            if (property.key.toLowerCase().startsWith("${prefix.toLowerCase()}.")) {
+                def key = property.key.toLowerCase().replace("${prefix.toLowerCase()}.", "").replace(".", "_")
+                fields_map[key] = property.value
+            }
+        }
+    }
+
+    // Then define the found properties as buildConfigFields
+    // Also define found properties that started with `res.` (`res_*` keys once in the `fields_map`) as resValues
+    fields_map.each {
+        variant.buildConfigField "String", it.key.toUpperCase(), "\"${it.value}\""
+        if (it.key.startsWith("res_")) {
+            variant.resValue "String", it.key.replace("res_", ""), "${it.value}"
+        }
```

Comment:
Note that I haven't seen any `wp.res.` property in our `gradle.properties` file so not sure why that was done in the first place, but since that logic was already here in the original code (see deleted code below) I figured I'd keep the option here.

Reply:
@AliSoftware Good catch, it looks like we don't have that anymore. I think we should use the opportunity to remove the `wp.res` logic.

Reply:
wfm 👍

Reply:
```suggestion
```